test/test_transforms_v2_refactored.py

```diff
@@ -1634,3 +1637,143 @@ def test_transform_negative_degrees_error(self):
+            # only 1 non-zero value is present, basically if the transform had
+            # no effect. Here we make sure that:
+            # - there is at least one sample with 2 non-zero values
+            # - there is no sample with more than 2 non-zero values
+            # When CutMix() and MixUp() are called in sequence together, we
+            # should expect 3 instead of 2. That's the
+            # expected_num_non_zero_labels threshold.
+            num_non_zero_values = (target != 0).sum(axis=-1)
+            assert (num_non_zero_values == expected_num_non_zero_labels).any()
+            assert (num_non_zero_values <= expected_num_non_zero_labels).all()
+            assert (num_non_zero_values > 0).all()  # Note: we already know that from target.sum(axis=-1) check above
+
+        # After Dataloader, as unpacked input
+        img, target = next(iter(dl))
+        assert target.shape == (batch_size, num_categories) if one_hot else (batch_size,)
+        img, target = cutmix_mixup(img, target)
+        check_output(img, target)
+
+        # After Dataloader, as packed input
+        packed_from_dl = next(iter(dl))
+        assert isinstance(packed_from_dl, list)
+        img, target = cutmix_mixup(packed_from_dl)
+        check_output(img, target)
+
+        # As collation function. We expect default_collate to be used by users.
+        def collate_fn_1(batch):
+            return cutmix_mixup(default_collate(batch))
+
+        def collate_fn_2(batch):
+            return cutmix_mixup(*default_collate(batch))
+
+        for collate_fn in (collate_fn_1, collate_fn_2):
+            dl = DataLoader(dataset, batch_size=batch_size, collate_fn=collate_fn)
+            img, target = next(iter(dl))
+            check_output(img, target)
+
+    @needs_cuda
+    @pytest.mark.parametrize("T", [transforms.Cutmix, transforms.Mixup])
+    def test_cpu_vs_gpu(self, T):
+        num_categories = 10
+        batch_size = 3
+        H, W = 12, 12
+
+        imgs = torch.rand(batch_size, 3, H, W).to("cuda")
+        labels = torch.randint(0, num_categories, (batch_size,)).to("cuda")
+        cutmix_mixup = T(alpha=0.5, num_categories=num_categories)
+
+        _check_kernel_cuda_vs_cpu(cutmix_mixup, input=(imgs, labels), rtol=None, atol=None)
+
+    @pytest.mark.parametrize("T", [transforms.Cutmix, transforms.Mixup])
+    def test_error(self, T):
+
+        num_categories = 10
+        batch_size = 9
+
+        # imgs = torch.randint(0, 256, (batch_size, 3, 12, 12), dtype=torch.uint8)
+        imgs = torch.rand(batch_size, 3, 12, 12)
+
+        for input_with_bad_type in (
+            F.to_pil_image(imgs[0]),
+            datapoints.Mask(torch.rand(12, 12)),
+            datapoints.BoundingBox(torch.rand(2, 4), format="XYXY", spatial_size=12),
+        ):
+            with pytest.raises(ValueError, match="does not support PIL images, "):
+                T(alpha=0.5)(input_with_bad_type)
```

Comment:
Since all errors are runtime errors below, can we just do `tranform = T(alpha=0.5)` upfront rather than instantiating it every time?